src/Licensing/Outpost.php

```diff
@@ -44,12 +45,25 @@ private function request()
             return $this->getCachedResponse();
         }
 
-        try {
-            return $this->performAndCacheRequest();
-        } catch (ConnectException $e) {
-            return $this->cacheAndReturnErrorResponse($e);
-        } catch (RequestException $e) {
-            return $this->handleRequestException($e);
+        $lock = $this->cache()->lock(static::LOCK_KEY, 5);
+
+        if ($lock->get()) {
+            // If we have a cached response from a previous lock, we can just return it.
+            if ($this->hasCachedResponse()) {
+                $lock->release();
+
+                return $this->getCachedResponse();
+            }
```

Comment:
Since we now have this "hasCachedResponse" guard here, can we get rid of the one above?